modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C

```diff
@@ -0,0 +1,326 @@
+  mooseAssert(_problem, "Problem pointer is null");
+
+  if (!_problem->isSNESMFReuseBaseSetbyUser())
+    _problem->setSNESMFReuseBase(false, false);
+
+  const std::vector<UserObjectName> user_object_vector =
+      getParam<std::vector<UserObjectName>>("user_object_physics");
+
+  for (const auto i : index_range(user_object_vector))
+  {
+    if (user_object_vector[i] == "GapFluxModelConduction")
+    {
+      auto var_params = _factory.getValidParams("GapFluxModelConduction");
+
+      var_params.set<std::vector<VariableName>>("temperature") =
+          getParam<std::vector<VariableName>>("temperature");
+      var_params.set<Real>("gap_conductivity") = getParam<Real>("gap_conductivity");
+
+      if (isParamValid("gap_conductivity_function"))
+        var_params.set<FunctionName>("gap_conductivity_function") =
+            getParam<FunctionName>("gap_conductivity_function");
+
+      if (isParamValid("gap_conductivity_function_variable"))
+        var_params.set<std::vector<VariableName>>("gap_conductivity_function_variable") =
+            getParam<std::vector<VariableName>>("gap_conductivity_function_variable");
+
+      var_params.set<Real>("min_gap") = getParam<Real>("min_gap");
+      var_params.set<unsigned int>("min_gap_order") = getParam<unsigned int>("min_gap_order");
+
+      var_params.set<std::vector<BoundaryName>>("boundary") =
+          getParam<std::vector<BoundaryName>>("boundary");
+
+      var_params.set<bool>("use_displaced_mesh") = true;
+
+      _problem->addUserObject("GapFluxModelConduction",
+                              "gap_flux_model_conduction_object_" + MooseUtils::shortName(name()) +
+                                  "_" +
+                                  Moose::stringify(thermal_action_userobject_conduction_counter++),
+                              var_params);
+    }
+    else if (user_object_vector[i] == "GapFluxModelRadiation")
+    {
+      auto var_params = _factory.getValidParams("GapFluxModelRadiation");
+
+      var_params.set<Real>("stefan_boltzmann") = getParam<Real>("stefan_boltzmann");
+      var_params.set<Real>("primary_emissivity") = getParam<Real>("primary_emissivity");
+      var_params.set<Real>("secondary_emissivity") = getParam<Real>("secondary_emissivity");
+
+      var_params.set<std::vector<BoundaryName>>("boundary") =
+          getParam<std::vector<BoundaryName>>("boundary");
+
+      var_params.set<std::vector<VariableName>>("temperature") =
+          getParam<std::vector<VariableName>>("temperature");
+
+      var_params.set<bool>("use_displaced_mesh") = true;
+
+      _problem->addUserObject("GapFluxModelRadiation",
+                              "gap_flux_model_radiation_object_" + MooseUtils::shortName(name()) +
+                                  "_" +
+                                  Moose::stringify(thermal_action_userobject_radiation_counter++),
+                              var_params);
+    }
+    else
+      paramError("user_object_physics",
```

Comment:
So you cannot use the new pressure dependent conduction?

Reply:
My suggestion would be to have the user provide the names of the UO blocks (that they add manually in the input). Passing in type names the way it seems to happen here is a bit of an antipattern (that duplicates the factory design).

Reply:
That's a particular model that we should debate whether it belongs to the action (see https://mooseframework.inl.gov/source/userobjects/GapFluxModelPressureDependentConduction.html#cincotti2007modeling). If it's going to be heavily used, we should add it.

Reply:
So you are suggesting the users add user objects in addition to the action? Hmm...

Reply:
You could add that as an option if you want bro put the basic models into the action. That way the users aren't hemmed in.

Reply:
I don't know what to think of this. In a way, I thought actions should "hem the users in". But sure, we could add another option.

Reply:
It is an extremely low effort addition to add an `other_gap_flux_models` parameter that gets appended to the list of user objects that are passed to the constraint. I disagree that actions "should hem in users". It is certainly a side effect of simplifying the syntax and removing options, but definitely not the goal. 

I think we can have both here. The basic set of gap flux models built in, and a simple option to add custom models (especially since those might come up in apps).

Reply:
> It is certainly a side effect of simplifying the syntax and removing options, but definitely not the goal.

Don't truly follow the side effect rationale. As such, it'd appear this is a code development exercise instead of achieving ease of use on the user side. 

Anyway, as I had said, we can have another option in the action.

Reply:
?